core_v2/sources/test_helper.move

```diff
@@ -82,14 +82,30 @@ module aptos_names_v2::test_helper {
         let is_subdomain = option::is_some(&subdomain_name);
 
         let user_balance_before = coin::balance<AptosCoin>(user_addr);
+
         let user_reverse_lookup_before = domains::get_reverse_lookup(user_addr);
+
         let maybe_target_address = domains::name_resolved_address(subdomain_name, domain_name);
         let name_reverse_lookup_before = if (option::is_some(&maybe_target_address)) {
-            domains::get_reverse_lookup(*option::borrow(&maybe_target_address))
+            let result = domains::get_reverse_lookup(*option::borrow(&maybe_target_address));
+            // When domain and subdomain both have target address set to user, and domain is primary name
+            // get_reverse_lookup returns the address of domain token
+            // we need to make sure the name token address match the name token address returnes from reverse lookup
+            // to know the name does have reverse look up (i.e. name is primary name)
+            if (result == option::some(domains::token_addr(domain_name, subdomain_name))) {
+                result
+            } else {
+                option::none()
+            }
```

Comment:
this is the most important change in this pr, after i set target address for subdomain the unit test keeps failing because it thinks `name_reverse_lookup_before` exist for subdomain now, actually it doesn't , `name_reverse_lookup_before` only be true when that name is a primary name, not when user has a primary name, that is what `user_reverse_lookup_before` for.